ygen/protogen.go

```diff
@@ -506,27 +584,141 @@ func genProto3Msg(msg *yangDirectory, msgs map[string]*yangDirectory, state *gen
+	definedFieldNames  map[string]bool           // definedFieldNames specifies the field names that have been defined in the context.
+	state              *genState                 //state is the current generator state.
+	cfg                *protoMsgConfig
+	parentPkg          string // parentPackage stores the name of the protobuf package that the field's parent is within.
+}
+
+// addProtoListField modifies the field definition in fieldDef (which must correspond to a list field of a
+// YANG schema) to contain the definition of the field described by the args. In the case that the list is keyed
+// and nested messages are being output, the generated protobuf message for the key is appended to the supplied
+// message definition (msgDef). If nested messages are not being output, a definition of the key message is returned.
+// Along with the optional key message, it returns a list of the imports being used for the list.
+func addProtoListField(fieldDef *protoMsgField, msgDef *protoMsg, args *protoDefinitionArgs) (*protoMsg, []string, util.Errors) {
+	listDef, keyMsg, err := protoListDefinition(args)
+	if err != nil {
+		return nil, nil, []error{fmt.Errorf("could not define list %s: %v", args.field.Path(), err)}
+	}
+
+	var nKeyMsg *protoMsg
+	if keyMsg != nil {
+		if args.cfg.nestedMessages {
+			// If nested messages are being output, we must ensure that the
+			// generated key message is output within the parent message - hence
+			// it is generated directly here and appended to the child messages.
+			kc, cerrs := genProto3MsgCode(args.parentPkg, []*protoMsg{keyMsg}, false)
+			if cerrs != nil {
+				return nil, nil, cerrs
+			}
+			msgDef.ChildMsgs = append(msgDef.ChildMsgs, kc)
+		} else {
+			nKeyMsg = keyMsg
+		}
+	}
+
+	fieldDef.Type = listDef.listType
+
+	// Lists are always repeated fields.
+	fieldDef.IsRepeated = true
+	return nKeyMsg, listDef.imports, nil
+}
+
+// addProtoContainerField modifies the field definition in fieldDef (which must correspond to a container field of
+// a YANG schema) to contain the definition of the field described by the args. It returns a slice of strings containing
+// the protobuf package imports that are required for the container defintion.
+func addProtoContainerField(fieldDef *protoMsgField, args *protoDefinitionArgs) ([]string, error) {
+	childmsg, ok := args.definedDirectories[args.field.Path()]
+	if !ok {
+		return nil, fmt.Errorf("proto: could not resolve %s into a defined struct", args.field.Path())
+	}
+
+	imports := map[string]interface{}{}
+
+	var pfx string
+	if !(args.cfg.compressPaths && args.directory.isFakeRoot) {
+		childpkg := args.state.protobufPackage(childmsg.entry, args.cfg.compressPaths)
+		// Add the import to the slice of imports if it is not already
+		// there. This allows the message file to import the required
+		// child packages.
+		childpath := importPath(args.cfg.baseImportPath, args.cfg.basePackageName, childpkg)
+		if _, ok := imports[childpath]; !ok {
```

Comment:
if imports[childpath] != nil

Reply:
Done -- `if imports[childpath] == nil` since this is `!ok`.